### DISCUSSION TITLE
`module.exports` multiple `onboardingConfig`

### DISCUSSION THREAD

**Discussion by BenB196** [renovatebot/renovate#21227] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.159.1-slim

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.8

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to have a single project, which Renovate runs from. In this project, I want to have a single `config.js` file. In this file, I want to be able to specify multiple `onboardingConfig` that leverage `repositories` to determine which `onboardingConfig` gets applied to which `repositories`.

The overall goal is to be able to more easily onboard projects differently based on the groups they are a part of.

### What you tried so far.

I have tried setting multiple `module.exports` in a single `config.js`

<details>
<summary>Example</summary>

```javascript
module.exports = {
  "autodiscover": true,
  "onboarding": true,
  "autodiscoverFilter": [
    "project-a"
  ],
  onboardingConfig: {
    extends: ['config:base'],
  }
}

module.exports = {
  "autodiscover": true,
  "onboarding": true,
  "autodiscoverFilter": [
    "project-b"
  ],
  onboardingConfig: {
    extends: ['config:base'],
   //something else
  }
}
```

</details>

In the above case, Renovate only used the last `module.exports`.

The other thing I tried was doing a branch per desired `module.exports`. This works, but is less than ideal, as now multiple branches need to be maintained.

**Comment by rarkins** [comment, marked as answer]
You can configure like this:

```
module.exports = {
  autodiscover: true,
  onboarding: true,
  repositories: [{
    repository: "org/project-a",
    onboardingConfig: {
      extends: ['config:base'],
    },
  }, {
    repository: "org/project-b",
    onboardingConfig: {
      extends: ['config:base', 'something/else'],
      // more
    },
  }]
}
```

Note:
- `repository` is an exact match, not a pattern
- Any repositories not explicitly configured in the `repositories` list will just get the default config

**Reply by BenB196** [reply]
Thanks @rarkins this seems to work, one question though:

> `repository` is an exact match, not a pattern

Are there any plans to make it support pattern matching, or a reason it doesn't? (I wasn't able to find an existing issue on the topic).

The reason I ask is to make things a bit easier, I might want to have something like:

```javascript
module.exports = {
  "autodiscover": true,
  "onboarding": true,
  "autodiscoverFilter": [
    "org-a/**",
    "org-b/**"
  ],
  repositories: [{
    repository: "org-a/**",
    onboardingConfig: {
      extends: ['config:base'],
    },
  }, {
    repository: "org-b/**",
    onboardingConfig: {
      extends: ['config:base', 'something/else'],
      // more
    },
  }]
}
```

**Reply by rarkins** [reply]
There's no plans, but nothing necessarily stopping it if you or someone else wanted to contribute it as a feature.

**Reply by BenB196** [reply]
Thanks for the information
